src/repositories/concrete/mongoTransactionRepository.ts

```diff
@@ -19,17 +20,20 @@ export class MongoTransactionRepository implements ITransactionRepository {
     public async getTransactionsByQueryAsync(queries: [string, string]): Promise<Transaction[]> {
         try {
             const database = await this.client.connectAsync();
-            const allTransactions: Transaction[] = await database.collection(this.tableName);
+            const allTransactions: Transaction[] = database.collection(this.tableName);
             let filteredTransactions: Transaction[] = new Array();
-            for (const transaction of allTransactions) {
-                for (const query of queries) {
-                    if (this.paramQueryEqualsToFieldName(query[0])) {
-                        if (transaction.query[0] === query[1]) {
-                            filteredTransactions.push(transaction);
-                        }
-                    }
+            
+            for (const query of queries) {
+                const field = allTransactions[0][query[0]];
```

Comment:
move the duplicate code to a separate class `RepositoryOperations` where these methods can live.